app/assets/javascripts/components/vm_cloud/vm-cloud-add-security-group-component.js

```diff
@@ -0,0 +1,61 @@
+ManageIQ.angular.app.component('vmCloudAddSecurityGroupComponent', {
+  bindings: {
+    recordId: '@',
+  },
+  controllerAs: 'vm',
+  controller: vmCloudAddSecurityGroupFormController,
+  templateUrl: '/static/vm_cloud/add_security_group.html.haml',
+});
+
+vmCloudAddSecurityGroupFormController.$inject = ['API', 'miqService'];
+
+function vmCloudAddSecurityGroupFormController(API, miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.afterGet = false;
+    vm.vmCloudModel = {
+      security_group: null,
+    };
+    vm.security_groups = [];
+    vm.formId = vm.recordId;
+    vm.model = "vmCloudModel";
+    vm.saveable = miqService.saveable;
+    miqService.sparkleOn();
+    API.get("/api/vms/" + vm.recordId).then(function(data) {
+      tenantId = data.cloud_tenant_id;
+      API.get("/api/vms/" + vm.recordId + "/security_groups?expand=resources&attributes=id,name").then(function(data) {
+        currentSecurityGroups = data.resources;      
+        API.get("/api/cloud_tenants/" + tenantId + "/security_groups?expand=resources&attributes=id,name").then(function(data) {
+          vm.security_groups = data.resources.filter(function (securityGroup) {
+            for (currentSecurityGroup of currentSecurityGroups) {
+              if (securityGroup.id == currentSecurityGroup.id) {
+                return false;
+              }
+            }
+            return true;
+          });
+          vm.afterGet = true;
+          vm.modelCopy = angular.copy( vm.vmCloudModel );
+        }).catch(miqService.handleFailure);
+      }).catch(miqService.handleFailure);
+    }).catch(miqService.handleFailure);
+    miqService.sparkleOff();
```

Comment:
Ah, and it doesn't make sense to call `sparkleOff` directly from `$onInit`, since it will happen right away, not after the data actually arrives .. this should be inside that last `then` instead.